### ISSUE TITLE
[v2 QUESTION]: Should We Unify the Prediction Output Format for Binary and Multiple Classification Tasks?

### ISSUE THREAD

**Issue by oscarwumit** [chemprop/chemprop#1181]
## What are you trying to do?
Currently, Chemprop’s output format for classification predictions differs between binary and multiclass tasks:

- **Binary Classification:**  
  The prediction output is a single probability value (e.g., `0.7` for "positive" class).

- **Multiclass Classification:**  
  The prediction output is provided as a string list containing the probabilities for each class (e.g., `[0.7, 0.2, 0.1]`).

This discrepancy can complicate downstream processing. However, because any change to the current output format will affect backward compatibility, I’d like to gauge the community’s interest in unifying the prediction output format.

## Potential Unifying Solution

I propose introducing new optional flags that enable a unified output format while maintaining backward compatibility:

1. **Unified Probability Output:**  
   For binary classification, instead of outputting a single probability value, output both class probabilities (e.g., `[P(class_0), P(class_1)]` with `P(class_1) = 1 – P(class_0)`). This makes the binary output consistent with the multiclass output (e.g., `[P(class_0), P(class_1)], P(class_2) ... ] `).

2. **Optional Predicted Class Label Column:**  
   Regardless of task type, include an extra column (e.g., `predicted_label`) that specifies the class corresponding to the highest probability.
   For binary classification, the default cutoff is assumed to be `0.5` (i.e., if `P(class_1) > 0.5` then predict class 1). We will leave it to users to adjust the cutoff in their downstream tasks if needed, meaning that the CSV generation in Chemprop will always use the 0.5 cutoff by default.

## Question

**Would the community be in favor of unifying the prediction output format as described above?** 
- Should we adopt the unified probability output for binary tasks so that they return a full probability distribution?
- Would you also find it useful to have an additional `predicted_label` column for both binary and multiclass tasks?

Any feedback or suggestions on this topic would be greatly appreciated.

**Comment by JacksonBurns** [comment]
Option 2 would nicely satisfy https://github.com/chemprop/chemprop/discussions/1159 by indirectly exposing the output activation function via the CLI - I really like that one

**Comment by m-aebrer** [comment]
Really a fan of this idea. A unified output format for all types of classification (binary vs multiclass) means that any downstream API I am supporting which uses chemprop models will just be _that_ much simpler to develop and maintain! I actually ran into an issue recently when trying to load a classification model, without realizing that binary classifiers are treated as a special case. It created some confusion for sure.

**Comment by KnathanM** [comment]
The origin of the difference between outputs for binary and multiclass goes back to their predictors in `chemprop/nn/predictors.py`. For binary, the FFN only predicts a single value that gets sigmoided. For multiclass, the FFN predicts as many values as there are classes and the values get softmaxed. I think API users can use `MulticlassClassificationFFN` with `n_classes=2` if they want binary classification with the desired output above.

For the CLI, I'm not against adding a flag for this to replicate the output of sklearn `predict_proba`. If someone opens a PR for this, please tag me. 

I think there also wouldn't be backward compatibility issues to add a new method to `BinaryClassificationFFN` and `MulticlassClassificationFFN` called `predict_proba` to copy sklearn. I think for the binary case it would look like 
```
def predict_proba(self, Z: Tensor) -> Tensor:
    p1 = self.forward(Z)
    return torch.stack((1 - p1, p1), dim=-1)
```
And for `MulticlassClassificationFFN` it would just be `predict_proba = forward`. 

If there is still interest in this, we could investigate this further.
